File: android/crewrp-core/src/main/kotlin/app/crewrp/core/TeamRole.kt
package app.crewrp.core

enum class TeamRole {
    ADMIN,
    MEMBER,
    NONE;

    val canEdit: Boolean get() = this == ADMIN

    companion object {
        const val ADMIN_SLUG = "admins"
        const val MEMBER_SLUG = "members"

        fun resolve(slugs: Collection<String>): TeamRole {
            val set = slugs.toSet()
            if (ADMIN_SLUG in set) return ADMIN
            if (MEMBER_SLUG in set) return MEMBER
            return NONE
        }
    }
}

data class Session(
    val org: String,
    val repo: String,
    val teamRole: TeamRole,
)

File: ios/Sources/CrewRPCore/TeamRole.swift
public enum TeamRole: String, Sendable, Equatable, Codable {
    case admin
    case member
    case none

    public static let adminSlug = "admins"
    public static let memberSlug = "members"

    public static func resolve(slugs: [String]) -> TeamRole {
        let set = Set(slugs)
        if set.contains(adminSlug) { return .admin }
        if set.contains(memberSlug) { return .member }
        return .none
    }

    public var canEdit: Bool { self == .admin }
}

public struct Session: Sendable, Equatable, Codable {
    public var org: String
    public var repo: String
    public var teamRole: TeamRole

    public init(org: String, repo: String, teamRole: TeamRole) {
        self.org = org
        self.repo = repo
        self.teamRole = teamRole
    }
}
